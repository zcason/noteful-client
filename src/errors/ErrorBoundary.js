import React from 'react';
import PropTypes from 'prop-types';

class ErrorBoundary extends React.Component {
    state = {
        hasError: false
    }
    static getDerivedStateFromError(error) {
        // Update state so the next render will show the fallback UI.
        return { hasError: true };
    }
    render() {
        if (this.state.hasError) {
            return (
                <h2>{this.props.message}</h2>
            );
        }
        return this.props.children;
    }
}

ErrorBoundary.propTypes = {
    message: PropTypes.string
}

export default ErrorBoundary;