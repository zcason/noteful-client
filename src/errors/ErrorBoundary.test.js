import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ErrorBoundary from './ErrorBoundary';

describe(`ErrorBoundary component`, () => {
    const props = {
      message: "Test Error Message"
    }
  
    it('renders an ErrorBoundary by default', () => {
      const wrapper = shallow(<ErrorBoundary />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('renders an error message with message when in props', () => {
      const msg = shallow(<ErrorBoundary {...props} />)
        .find('Test Error Message')
      expect(toJson(msg)).toMatchSnapshot()
    })
  })
  