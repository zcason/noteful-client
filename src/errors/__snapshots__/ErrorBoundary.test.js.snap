// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component renders an ErrorBoundary by default 1`] = `""`;

exports[`ErrorBoundary component renders an error message with message when in props 1`] = `null`;
